newton/_src/usd/utils.py

```
@@ -745,31 +854,63 @@ def get_mesh(
-                normals_fv = normals[normal_indices]  # (C,3) expanded
+        if normals_interpolation == UsdGeom.Tokens.faceVarying:
+            prim_path = str(prim.GetPath())
+            if normal_indices is not None and len(normal_indices) > 0:
+                normals_fv = _expand_indexed_primvar(normals, normal_indices, "Normal", prim_path)
             else:
                 # If faceVarying, values length must match number of corners
                 if len(normals) != len(indices):
                     raise ValueError(
-                        f"Length of normals ({len(normals)}) does not match length of indices ({len(indices)}) for mesh {prim.GetPath()}"
+                        f"Length of normals ({len(normals)}) does not match length of indices ({len(indices)}) for mesh {prim_path}"
                     )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)